pkg/abstractions/endpoint/instance.go

```diff
@@ -31,21 +31,30 @@ type endpointInstance struct {
 }
 
 func (i *endpointInstance) startContainers(containersToRun int) error {
+	secrets, err := abstractions.ConfigureContainerRequestSecrets(i.Workspace, *i.buffer.stubConfig)
+	if err != nil {
+		return err
+	}
+
+	env := []string{
+		fmt.Sprintf("BETA9_TOKEN=%s", i.Token.Key),
+		fmt.Sprintf("HANDLER=%s", i.StubConfig.Handler),
+		fmt.Sprintf("ON_START=%s", i.StubConfig.OnStart),
+		fmt.Sprintf("STUB_ID=%s", i.Stub.ExternalId),
+		fmt.Sprintf("STUB_TYPE=%s", i.Stub.Type),
+		fmt.Sprintf("CONCURRENCY=%d", i.StubConfig.Concurrency),
+		fmt.Sprintf("KEEP_WARM_SECONDS=%d", i.StubConfig.KeepWarmSeconds),
+		fmt.Sprintf("PYTHON_VERSION=%s", i.StubConfig.PythonVersion),
+		fmt.Sprintf("TIMEOUT=%d", i.StubConfig.TaskPolicy.Timeout),
+	}
+
+	env = append(env, secrets...)
+
 	for c := 0; c < containersToRun; c++ {
 		containerId := i.genContainerId()
 		runRequest := &types.ContainerRequest{
 			ContainerId: containerId,
-			Env: []string{
```

Comment:
We may want to append the secrets prior to our internal env vars to prevent a user from overriding something critical by mistake